plugins/base/src/main/kotlin/translators/descriptors/DefaultDescriptorToDocumentableTranslator.kt

```diff
@@ -784,36 +784,45 @@ private class DokkaDescriptorVisitor(
+        mapNotNull { it.toAnnotation() }.singleOrNull { it.dri.classNames == "ParameterName" }?.params?.get("name")
             .safeAs<StringValue>()?.value?.let { unquotedValue(it) }
 
-    private suspend fun KotlinType.toBound(): Bound = when (this) {
-
-        is DynamicType -> Dynamic
-        is AbbreviatedType -> TypeAliased(
-            abbreviation.toBound(),
-            expandedType.toBound()
-        )
-        else -> when (val ctor = constructor.declarationDescriptor) {
-            is TypeParameterDescriptor -> TypeParameter(
-                dri = DRI.from(ctor),
-                name = ctor.name.asString(),
-                presentableName = annotations.getPresentableName()
-            )
-            is FunctionClassDescriptor -> FunctionalTypeConstructor(
-                DRI.from(ctor),
-                arguments.map { it.toProjection() },
-                isExtensionFunction = isExtensionFunctionType || isBuiltinExtensionFunctionalType,
-                isSuspendable = isSuspendFunctionTypeOrSubtype,
-                presentableName = annotations.getPresentableName()
+    private suspend fun KotlinType.toBound(): Bound {
+        suspend fun <T : Annotatable> annotations(): PropertyContainer<T> =
+            getAnnotations().takeIf { it.isNotEmpty() }?.let { annotations ->
+                PropertyContainer.withAll(annotations.toSourceSetDependent().toAnnotations())
+            } ?: PropertyContainer.empty()
+
+        return when (this) {
+            is DynamicType -> Dynamic
+            is AbbreviatedType -> TypeAliased(
+                abbreviation.toBound(),
+                expandedType.toBound()
             )
-            else -> GenericTypeConstructor(
-                DRI.from(ctor!!), // TODO: remove '!!'
-                arguments.map { it.toProjection() },
-                annotations.getPresentableName()
-            )
-        }.let {
-            if (isMarkedNullable) Nullable(it) else it
+            else -> when (val ctor = constructor.declarationDescriptor) {
+                is TypeParameterDescriptor -> TypeParameter(
+                    dri = DRI.from(ctor),
+                    name = ctor.name.asString(),
+                    presentableName = annotations.getPresentableName(),
+                    extra = annotations()
+                )
+                is FunctionClassDescriptor -> FunctionalTypeConstructor(
+                    DRI.from(ctor),
+                    arguments.map { it.toProjection() },
+                    isExtensionFunction = isExtensionFunctionType || isBuiltinExtensionFunctionalType,
+                    isSuspendable = isSuspendFunctionTypeOrSubtype,
+                    presentableName = annotations.getPresentableName(),
+                    extra = annotations()
+                )
+                else -> GenericTypeConstructor(
+                    DRI.from(ctor!!), // TODO: remove '!!'
```

Comment:
The `!!` can be removed here IMHO

Reply:
I dont think so since it is nullable and afaik we dont have a nullable constructor for DRI